Detectors/ITSMFT/test/HitAnalysis/src/HitAnalysis.cxx

```diff
@@ -225,12 +224,11 @@ void HitAnalysis::ProcessChips()
 
 void HitAnalysis::ProcessHits()
 {
-  for (TIter pointiter = TIter(mHitsArray).Begin(); pointiter != TIter::End(); ++pointiter) {
-    Hit *p = static_cast<Hit *>(*pointiter);
-    auto loc = p->GetPos();
-    auto locS = p->GetPosStart();
-    auto glo = mGeometry->getMatrixL2G(p->GetDetectorID())(loc);
-    auto gloS = mGeometry->getMatrixL2G(p->GetDetectorID())(locS);
+  for (auto& hit : *mHits) {
+    auto loc = hit.GetPos();
```

Comment:
It looks like i've introduced a bug here, could you please change the lines 228-231 to
auto loc  = mGeometry->getMatrixL2G^(hit.GetDetectorID())(hit.GetPos()); // global->local end position
auto locS = mGeometry->getMatrixL2G^(hit.GetDetectorID())(hit.GetPosStart()); // global->local start position

Reply:
will do

Reply:
sorry, my fault, misplaced (), should be
auto loc = mGeometry->getMatrixL2G(hit.GetDetectorID())^(hit.GetPos()); // global->local end position
auto locS = mGeometry->getMatrixL2G(hit.GetDetectorID())^(hit.GetPosStart()); // global->local start position